### PR TITLE
do not call absl::InitializeSymbolizer on windows

### LINKED ISSUE #42690: Ray fails on Windows with torchaudio (simple 3 line repro)
### What happened + What you expected to happen

ray.init() failing in Windows on a CUDA machine. Tried installing barebones CPU-only torchaudio (`pip install torchaudio`) but still fails. Repro with this:

```
import torchaudio
import ray
ray.init()
```

And then it crashes:

```
(ai) C:\projects\UntitledAI>python
Python 3.11.7 | packaged by Anaconda, Inc. | (main, Dec 15 2023, 18:05:47) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import ray
>>> import torchaudio
>>> ray.init()
2024-01-24 17:40:19,380 INFO worker.py:1715 -- Started a local Ray instance. View the dashboard at http://127.0.0.1:8265
[symbolize_win32.inc : 53] RAW: SymInitialize() failed: 87
Fatal Python error: Aborted

Stack (most recent call first):
  File "C:\Users\Bart\anaconda3\envs\ai\Lib\site-packages\ray\_private\worker.py", line 2342 in connect
  File "C:\Users\Bart\anaconda3\envs\ai\Lib\site-packages\ray\_private\worker.py", line 1726 in init
  File "C:\Users\Bart\anaconda3\envs\ai\Lib\site-packages\ray\_private\client_mode_hook.py", line 103 in wrapper
  File "<stdin>", line 1 in <module>

Extension modules: msgpack._cmsgpack, google._upb._message, psutil._psutil_windows, setproctitle, yaml._yaml, charset_normalizer.md, ray._raylet, numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, torch._C, torch._C._fft, torch._C._linalg, torch._C._nested, torch._C._nn, torch._C._sparse, torch._C._special, _cffi_backend, multidict._multidict, yarl._quoting_c, aiohttp._helpers, aiohttp._http_writer, aiohttp._http_parser, aiohttp._websocket, frozenlist._frozenlist, grpc._cython.cygrpc (total: 36)
```

Seeing this in a larger project, too, but this seems to be the minimum repro.

### Versions / Dependencies

Python 3.11.7
Windows 10

Packages installed:
aiosignal==1.3.1
attrs==23.2.0
certifi==2023.11.17
charset-normalizer==3.3.2
click==8.1.7
colorama==0.4.6
filelock==3.13.1
frozenlist==1.4.1
fsspec==2023.12.2
idna==3.6
Jinja2==3.1.3
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
MarkupSafe==2.1.4
mpmath==1.3.0
msgpack==1.0.7
networkx==3.2.1
packaging==23.2
protobuf==4.25.2
PyYAML==6.0.1
ray==2.9.1
referencing==0.32.1
requests==2.31.0
rpds-py==0.17.1
sympy==1.12
torch==2.1.2
torchaudio==2.1.2
typing_extensions==4.9.0
urllib3==2.1.0

### Reproduction script

import torchaudio
import ray
ray.init()

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by mattip**
## Why are these changes needed?

Calling `absl::InitializeSymbolizer` twice will crash on windows. Other libraries, like pytorchaudio, call it on import. So ray should not call it. The function is "nice to have", it gives a better traceback on crashes.

## Related issue number
Closes #42690

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by mattip** [comment]
The core redis test CI run failed in test_runtime_env_working_dir_3, which appears on the [flaky.io](https://flaky-tests.ray.io/). All the windows CI runs passed.

File: src/ray/util/logging.cc [hunk 1]
```diff
@@ -383,6 +383,9 @@ void RayLog::InstallFailureSignalHandler(const char *argv0, bool call_previous_h
   if (is_failure_signal_handler_installed_) {
     return;
   }
+  // InitializeSymbolizer cannot be called twice, and is called in
+  // other libraries like pytorchaudio. So do not call it here.
+  // absl::InitializeSymbolizer(argv0);
   absl::InitializeSymbolizer(argv0);
```
Comment (jjyao) [inline src/ray/util/logging.cc:389]: This is still called?

Is there a way to check if `InitializeSymbolizer` is called and skip calling it again if it's already called.

Comment (mattip) [reply src/ray/util/logging.cc]: Ahh, sorry, fixed. 
> Is there a way to check if InitializeSymbolizer is called 

I could not find one, I extended the comment. Also: this is a nice-to-have feature, not required for running. I think it makes sense to avoid it since other projects may call it again after ray initializes, and we have no control over that.

File: src/ray/util/logging.cc [hunk 2]
```diff
@@ -383,7 +383,11 @@ void RayLog::InstallFailureSignalHandler(const char *argv0, bool call_previous_h
   if (is_failure_signal_handler_installed_) {
     return;
   }
-  absl::InitializeSymbolizer(argv0);
+  // InitializeSymbolizer cannot be called twice, (causes a crash)and is called
+  // in other libraries like pytorchaudio. It does not seem there is a API to
+  // determine if it has already been called, and is only needed to provide
+  // better stack traces on crahses. So do not call it here.
+  // absl::InitializeSymbolizer(argv0);
```
Comment (jjyao) [inline src/ray/util/logging.cc:390]: This completely disable it even for non-windows?

Comment (mattip) [reply src/ray/util/logging.cc]: Good point. Fixed to exclude windows only.

**Review by jjyao — APPROVED** [review]
